main.py

```diff
@@ -19,67 +17,86 @@
 
 sniped = {}
 sniped_edit = {}
-
 #These dictionaries store the deleted messages that our on_message events capture
 
-
-
 class myBot(commands.Bot): #Being used as an event handler, but this class isn't needed for the bot to work overall (You'd just have to make the necessary changes). It's just more organized.
   async def on_ready(self):
       print(f"Logged in as {bot.user} (Version: {discord.__version__}") #When the bot is online this function prints.
 
-  async def on_message_delete(self, message): #When a message is deleted it looks at the channel that it happened in, the user who deleted it, (along with the other specifications) and appends it to the "sniped" dictionary.
-      sniped[message.channel.id] = [
-          message.content, message.author, message.attachments,
-          message.channel.name, message.created_at
-      ]
-
-  async def on_message_edit(self, before, after): #Same concept as on_message_delete
-      sniped_edit[before.channel.id] = [
-          before.content, before.author, before.channel.name,
-          before.created_at
-      ]
+  async def on_message_delete(self, message):
+        if message.author == self.user:
+            return # Ignore if the message was deleted by the bot itself
+        
+        sniped[message.channel.id] = [
+            message.content, message.author, message.attachments, message.stickers, message.channel.name, message.created_at
+        ]
 
+  async def on_message_edit(self, before, after):
+        sniped_edit[before.channel.id] = [
+           before.content, before.author, after.content, before.channel.name, before.created_at
+]
 
 bot = myBot(command_prefix=prefix, intents=intents)
 
-
-@bot.command()
-async def snipe(ctx):
+@bot.event
+async def on_ready():
+  print("Bot is up and ready!")
+  try:
+    synced = await bot.tree.sync()
+    print(f"Synced {len(synced)} command(s)") #prints how many commands the bot sees
+  except Exception as e:
+    print(e)   
+
+@bot.tree.command(name="snipe", description="will snipe the last message deleted")
+async def snipe(interaction: discord.Interaction):
+  try:
+    contents, target, attch, stickers, channel, time = sniped[interaction.channel.id]
+  except KeyError:
+    return await interaction.response.send_message("Nothing to snipe")
+
+  snipe_em = discord.Embed(description=contents, color=discord.Color.blurple(), timestamp=time) #this is the embeded message that the bot will send
+  snipe_em.set_author(name=target.name, icon_url=target.display_avatar.url) #change ".name" to "display_name" if you want the bot to show the name and not the discord @
+
+  attachment_url = None
+  if attch:  # If an attachment is found (img/video) then it will adjust the embed accordingly.
+    attachment = attch[0]
+    if attachment.proxy_url.endswith(('mp4', 'mov')):
+      attachment_url = attachment.proxy_url
+      snipe_em.description += f"\n Video found! link posted below" #when a video is found it will send this message
```

Comment:
Having "Video link found" being sent in the embed is kind of confusing so either
send it separately outside the embed or just remove it